packages/engine/Source/Scene/WebMapServiceImageryProvider.js

```diff
@@ -340,6 +341,9 @@ function WebMapServiceImageryProvider(options) {
     ),
     enablePickFeatures: options.enablePickFeatures,
   });
+
+  this._ready = true;
+  this._readyPromise = Promise.resolve(true);
```

Comment:
I may have missed something, but can you explain what these private properties are for? I don't see them used internally, and the public `.ready` and `.readyPromise` properties point to the relevant properties of UrlTemplateImageryProvider

Reply:
Sure!

So since the constructor method and ready properties are being deprecated, they still need to work before they are entirely removed. The `ready` and `readyPromise` properties are part of the public API, but they are also used internally for rendering imagery and terrain (ie https://github.com/CesiumGS/cesium/pull/11075#discussion_r1098820299 for terrain providers, https://github.com/CesiumGS/cesium/pull/11075/files#diff-da7b68dacf6bee3b99070639c851312908f93636cd7d4a0372ba03dfc254f6b5R244 for imagery providers). In order to avoid printing the depreciation warning from within internal code that a user may not have touched, we're now using the private versions of these properties.

Reply:
Thanks, that makes sense! I hadn't seen the usage in `GlobeSurfaceTileProvider`